### PR TITLE
[prometheus-postgres-exporter] Implement multiple target with /probe …

### PR DESCRIPTION
**Pull request by festeveira**
…endpoint

#### What this PR does / why we need it
This PR allows the user to use the multiple target support implemented by the exporter. It allows defining several postgres database targets, using the same or different auth modules. 

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [X] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [X] Chart Version bumped
- [X] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by zanhsieh — APPROVED** [review]

**Review by zeritti — CHANGES_REQUESTED** [review]
Thank you, @festeveira, for your PR. Please, see my comments below.

File: charts/prometheus-postgres-exporter/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -12,6 +12,52 @@ metadata:
...
+          targetLabel: target
+      {{- if $.Values.serviceMonitor.metricRelabelings -}}
+        {{ toYaml $.Values.serviceMonitor.metricRelabelings | nindent 8 }}
+      {{- end }}
+  {{- if $.Values.serviceMonitor.relabelings }}
+      relabelings: {{ toYaml $.Values.serviceMonitor.relabelings | nindent 8 }}
+  {{- end }}
+      params:
+        target:
+          - {{ .endpoint }}:{{ .port | default 5432 }}/postgres
```
[the remark this answers is not in the document]
Comment (festeveira) [reply charts/prometheus-postgres-exporter/templates/servicemonitor.yaml:50]: I'm sorry, I was not quite sure about this one. Would you implement it as a parametrizable value? From the exporter docs I'm not sure how it behaves when you point to a different database. I get the same results experimenting.

[the remark this answers is not in the document]
Comment (zeritti) [reply charts/prometheus-postgres-exporter/templates/servicemonitor.yaml:50]: The path should not be there. Consider scraping instance pg01 by means of the exporter through a URL where you'd set 

```bash
/probe?target=pg01:5432&auth_module=pg01
```

The _target_ is just a PostgreSQL instance details for which will be read out of the corresponding module.

**Review by zeritti — CHANGES_REQUESTED** [review]

File: charts/prometheus-postgres-exporter/templates/servicemonitor.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (festeveira) [reply charts/prometheus-postgres-exporter/templates/servicemonitor.yaml:50]: I understand. But, in my setup, I'm monitoring three different postgresql instances, all with different versions, 9.4, 14.2 and 15, and if I request the probe endpoint like you say I get `pg_up 0` and no other metrics. I always need to define a path with a valid database. Am I doing something wrong? Can you define the path in the `auth_modules`?

**Review by zanhsieh — APPROVED** [review]

**Review by zeritti — APPROVED** [review]
Thank you, @festeveira, LGTM.

**Pull request merged**